### ISSUE TITLE
TypeLibrary.Create crashes game trying to make a Dictionary

### ISSUE THREAD

**Issue by lotuspar** [sboxgame/issues#3849]
### Describe the bug

Title - game crashes instantly

### To Reproduce

```cs
TypeLibrary.Create<object>( typeof(Dictionary<string, string>) ); // crash
```

### Expected behavior

No crash

**Comment by timmybo5** [comment]
Why would you even need this? TypeLibrary is meant for reflection purposes with attributes, not to instantiate base types.
`var myDict = new Dictionary<string, string>();`

**Comment by lotuspar** [comment]
yes i know that (i am not stupid) it's an example to repro the bug

**Comment by peter-r-g** [comment]
This happens for any generic type that is not stored in the TypeLibrary (Related issue #2550). The TypeLibrary's IsAllowedType method infinitely recurses trying to verify the same generic type and stack overflows
